lnwallet/chainfee/log.go

```diff
@@ -27,3 +27,19 @@ func DisableLog() {
 func UseLogger(logger btclog.Logger) {
 	log = logger
 }
+
+// logClosure is used to provide a closure over expensive logging operations so
+// don't have to be performed when the logging level doesn't warrant it.
+type logClosure func() string
+
+// String invokes the underlying function and returns the result.
+func (c logClosure) String() string {
+	return c()
+}
+
+// newLogClosure returns a new closure over a function that returns a string
+// which itself provides a Stringer interface so that it can be used with the
+// logging system.
+func newLogClosure(c func() string) logClosure {
+	return logClosure(c)
+}
```

Comment:
For some reason I thought this was defined elsewhere in the codebase already.

Reply:
Yeah so in that case I'm gonna say we need to extract it to its own log package or something. We should not be duplicating this type of stuff everywhere.

Reply:
To be clear, my approval is pretty much only blocked on this.

Reply:
append a new commit to address this - think we got roughly 300 lines of code removed so kinda nice.

Reply:
It's beautiful 🥲